app/client/src/sagas/DatasourcesSagas.ts

```
@@ -1561,66 +1561,101 @@ function* fetchDatasourceStructureSaga(
+            source: {
+              id: action.payload.id,
+              name: datasource.name,
+              type: ENTITY_TYPE.DATASOURCE,
+            },
+          });
+        }
+
         // TODO: Fix this the next time the file is edited
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        errorMessage = (response.data as any).error?.message;
+        if (!!(response.data as any)?.error) {
```

Biome (1.9.4)

[error] Avoid redundant double-negation.

It is not necessary to use double-negation when a value will already be coerced to a boolean.
Unsafe fix: Remove redundant double-negation


(lint/complexity/noExtraBooleanCast)